app/importers/file_importers/student_section_grades_importer.rb

```diff
@@ -0,0 +1,31 @@
+class StudentSectionGradesImporter < Struct.new :school_scope, :client, :log, :progress_bar
+  def initialize(*)
+    super
+    self.student_lasid_map = Student.all.pluck(:local_id,:id).to_h
+    self.section_number_map = ActiveRecord::Base.connection.execute("select CONCAT(sections.section_number, '|', sections.term_local_id, '|', schools.local_id, '|', courses.course_number), sections.id  from sections inner join courses on courses.id = sections.course_id inner join schools on schools.id = courses.school_id").values.to_h
+  end
```

Comment:
To clarify my understanding, this is because Aspen doesn't have a primary key for sections.

Given that's the case, is the tuple of `(section_number, course_number, term_local_id)` not unique?  if we also need the school `local_id` to disambiguate, then should that be updated in the unique constraints on the `Section` model?

Reply:
Correct but not sure about adding it to the unique constraints of the Section model? The importer enforces the uniqueness. Is it possible to include a field from a parent table in the uniqueness constraint of a model?

Reply:
Ah my bad, the Section uniqueness constraints work because on the `Section` model the Insights `course_id` contains both the (course_number, school local_id) bits.

As for uniqueness constraints across tables, you can definitely do this in Rails/ActiveRecord if not at the database level, which I think would be good to capture.  But here `course_id` does that so it's already covered.